apps/api/src/http/v3/gql/mutations/reports.rs

```diff
@@ -274,6 +204,45 @@ impl ReportsMutation {
 				.ok_or(ApiError::NOT_FOUND)
 				.map_err(TransactionError::custom)?;
 
+			if let Some(new_priority) = new_priority {
+				tx.register_event(EventPayload {
+					actor_id: Some(auth_sesion.user_id()),
+					data: EventPayloadData::Ticket {
+						after: ticket.clone(),
+						data: EventTicketData::ChangePriority {
+							old: ticket.priority.clone(),
+							new: new_priority,
+						},
+					},
+					timestamp: chrono::Utc::now(),
+				})?;
+			}
+
+			if let Some(new_open) = new_open {
+				tx.register_event(EventPayload {
+					actor_id: Some(auth_sesion.user_id()),
+					data: EventPayloadData::Ticket {
+						after: ticket.clone(),
+						data: EventTicketData::ChangeOpen {
+							old: ticket.open,
+							new: new_open,
+						},
+					},
+					timestamp: chrono::Utc::now(),
+				})?;
+			}
+
+			if let Some(event_ticket_data) = event_ticket_data {
+				tx.register_event(EventPayload {
+					actor_id: Some(auth_sesion.user_id()),
+					data: EventPayloadData::Ticket {
+						after: ticket.clone(),
+						data: event_ticket_data,
+					},
+					timestamp: chrono::Utc::now(),
+				})?;
+			}
+
```

Comment:
Is this truly what we want to do?

What if we only emit a single event per / mutation?


And we combine these multiple events into a single?

Reply:
I tried to keep it as simple as possible. What would be the advantage of grouping multiple events into one?